govcd/multi_site_test.go

```diff
@@ -0,0 +1,366 @@
+//go:build system || functional || ALL
+
+/*
+ * Copyright 2024 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+package govcd
+
+import (
+	"fmt"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	. "gopkg.in/check.v1"
+	"os"
+	"strings"
+	"time"
+)
+
+// #nosec G101 -- These credentials are fake for testing purposes
+const (
+	secondVcdUrl              = "VCD_URL2"
+	secondVcdUser             = "VCD_USER2"
+	secondVcdPassword         = "VCD_PASSWORD2"
+	secondVcdSysOrg           = "VCD_SYSORG2"
+	secondVcdOrg2             = "VCD_ORG2"
+	secondVcdOrgUser2         = "VCD_ORGUSER2"
+	secondVcdOrgUserPassword2 = "VCD_ORGUSER_PASSWORD2"
+)
+
+func getClientConnectionFromEnv() (*VCDClient, error) {
+	vcdUrl := os.Getenv(secondVcdUrl)
+	user := os.Getenv(secondVcdUser)
+	password := os.Getenv(secondVcdPassword)
+	orgName := os.Getenv(secondVcdSysOrg)
+	if !strings.HasSuffix(vcdUrl, "/api") {
+		return nil, fmt.Errorf("the VCD URL must terminate with '/api'")
+	}
+	var missing []string
+	if vcdUrl == "" {
+		missing = append(missing, secondVcdUrl)
+	}
+	if user == "" {
+		missing = append(missing, secondVcdUser)
+	}
+	if password == "" {
+		missing = append(missing, secondVcdPassword)
+	}
+	if orgName == "" {
+		missing = append(missing, secondVcdSysOrg)
+	}
+	if len(missing) > 0 {
+		return nil, fmt.Errorf("missing environment variables for connection: %v", missing)
+	}
+
+	return newUserConnection(vcdUrl, user, password, orgName, true)
+}
+
+/*
+	Test_SiteAssociations will test the associations between two sites
+	To run this test, make a shell script like the one below, filling the variables
+	in addition to the VCD defined in govcd_test_config.yaml
+
+$ cat connection.sh
+export VCD_URL2=https://some-vcd-url.com/api
+export VCD_USER2=administrator
+export VCD_PASSWORD2='myPassword'
+export VCD_SYSORG2=System
+export VCD_ORG2=orgname2
+export VCD_ORGUSER2=org-admin-name
+export VCD_ORGUSER_PASSWORD2='myOrgAdminPassword'
+
+$	source connection.sh
+$ go test -tags functional -check.f Test_SiteAssociations -vcd-skip-vapp-creation -check.vv -timeout 0
+*/
+func (vcd *TestVCD) Test_SiteAssociations(check *C) {
+
```

Comment:
Similar to other place - should the test be skipped on VCD < 10.4.2 as per definition of [GetAllOrgs](https://github.com/vmware/go-vcloud-director/pull/669/files#diff-d48dd6aa67017584bb6e37e8e459a46748e9f2660ff686da9a84b3bb18449792R146)

Reply:
Done